src/testRunner/externalCompileRunner.ts

```diff
@@ -106,6 +106,65 @@ ${stripAbsoluteImportPaths(result.stderr.toString().replace(/\r\n/g, "\n"))}`;
     }
 }
 
+class DockerfileRunner extends ExternalCompileRunnerBase {
+    readonly testDir = "tests/cases/docker/";
+    kind(): TestRunnerKind {
+        return "docker";
+    }
+    initializeTests(): void {
+        // Read in and evaluate the test list
+        const testList = this.tests && this.tests.length ? this.tests : this.enumerateTestFiles();
+
+        // tslint:disable-next-line:no-this-assignment
+        const cls = this;
+        describe(`${this.kind()} code samples`, function(this: Mocha.ISuiteCallbackContext) {
+            this.timeout(cls.timeout); // 20 minutes
+            before(() => {
+                cls.exec("docker", ["build", ".", "-t", "typescript/typescript"], { cwd: Harness.IO.getWorkspaceRoot() }); // cached because workspace is hashed to determine cacheability
+            });
+            for (const test of testList) {
+                const directory = typeof test === "string" ? test : test.file;
+                const cwd = path.join(Harness.IO.getWorkspaceRoot(), cls.testDir, directory);
+                it(`should build ${directory} successfully`, () => {
+                    const imageName = `tstest/${directory}`;
+                    cls.exec("docker", ["build", "--no-cache", ".", "-t", imageName], { cwd }); // --no-cache so the latest version of the repos referenced is always fetched
+                    const cp: typeof import("child_process") = require("child_process");
+                    Harness.Baseline.runBaseline(`${cls.kind()}/${directory}.log`, cls.report(cp.spawnSync(`docker`, ["run", imageName], { cwd, timeout: cls.timeout, shell: true })));
+                });
+            }
+        });
+    }
+
+    private timeout = 1_200_000; // 20 minutes;
+    private exec(command: string, args: string[], options: { cwd: string, timeout?: number }): void {
+        const cp: typeof import("child_process") = require("child_process");
+        const stdio = isWorker ? "pipe" : "inherit";
+        const res = cp.spawnSync(command, args, { timeout: this.timeout, shell: true, stdio, ...options });
+        if (res.status !== 0) {
+            throw new Error(`${command} ${args.join(" ")} for ${options.cwd} failed: ${res.stderr && res.stderr.toString()}`);
+        }
+    }
+    report(result: ExecResult) {
+        // tslint:disable-next-line:no-null-keyword
+        return result.status === 0 && !result.stdout.length && !result.stderr.length ? null : `Exit Code: ${result.status}
+Standard output:
+${stripAbsoluteImportPaths(sanitizeTimestamps(result.stdout.toString().replace(/\r\n/g, "\n").replace(/\x1b\[[0-9;]*[a-zA-Z]/g, "")))}

```

Comment:
those last two regexes should be dumped into `sanitizeTimestamps` too (and its name should probably just be `sanitize`